plugins/integration/providers/voicebase/lib/kVoicebaseFlowManager.php

```diff
@@ -31,6 +31,7 @@ public function updatedJob(BatchJob $dbBatchJob)
 		$spokenLanguage = $providerData->getSpokenLanguage();
 
 		$transcripts = kTranscriptHelper::getAssetsByLanguage($entryId, array(TranscriptPlugin::getAssetTypeCoreValue(TranscriptAssetType::TRANSCRIPT)), $spokenLanguage);
+		$externalId = $entryId . '_' . $dbBatchJob->getId();
```

Comment:
Since $externalId  is built the same way in 3 different functions, its better adding one small function that builds it something like getExternalId($entryId , $jobId)  and use it in all the 3.